src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,346 @@
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4 67 8 $6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4 6 7 8 $6$7$8$9"));
+        }
+
+        List<String> specialChars = getSpecialCharacters("\\");
+        specialChars.removeAll(List.of("\\.", "\\-"));
+
+        specialChars.forEach(
+                specialChar -> wrongPhoneNumbers.forEach(
+                        number -> {
+                            tempList.add(
+                                    number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
+                                            "$1$2$3$4$5".concat(specialChar).concat("$7$8$9")));
+                            String twoDigits = number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$7");
+                            tempList.add(
+                                    number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
+                                            "$1$2$3$4$5$6".concat("\\".concat(twoDigits.replaceAll("^\\d", specialChar))).concat("$8$9")));
+                        }));
+
+        wrongPhoneNumbers.clear();
+        wrongPhoneNumbers.addAll(tempList);
+
+        return wrongPhoneNumbers;
+    }
+
+    private static List<String> createCorrectUserNames() {
+        List<String> correctNames = new ArrayList<>(List.of(
+                "pushkin",
+                "ThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100Symbols",
+                "1"));
+        List<String> specialChars = getSpecialCharacters("");
+        specialChars.addAll(getRussianAndUkrLetters());
+
+        correctNames.addAll(specialChars);
+
+        return correctNames;
+    }
+
+    private static List<String> createWrongUserNames() {
+
+        return List.of(
+                "",
+                " ",
+                "aaa bbb",
+                " aaa",
+                "aaa ");
+    }
+
+    private static List<String> createCorrectWordsEmptyOrMin2Max50() {
+        List<String> correctWords = new ArrayList<>(List.of(
+                "aZ",
+                "99",
+                "Za0",
+                "wW1-",
+                "wW2-w",
+                "wW3'",
+                "wW4'w",
+                "wW5`",
+                "wW6`w",
+                "wW7_",
+                "wW8_w"
+        ));
+
+        List<String> tempList = new ArrayList<>();
+        correctWords.forEach(word -> getRussianAndUkrLetters().forEach(
+                letter -> tempList.add(word.concat(letter))
+        ));
+
+        correctWords.addAll(tempList);
+        correctWords.addAll(List.of(
+                "",
+                "LengthIs50LengthIs50LengthIs50LengthIs50LengthIs50"));
+
+        return correctWords;
+    }
+
+    private static List<String> createWrongWordsEmptyOrMin2Max50() {
+        String correctWord = "a1A";
+
+        List<String> wrongWords = new ArrayList<>(List.of(
+                "w",
```

Comment:
Make such list with loop and 'w', 'W', ' ', '1' characters that's position would be changed.

Reply:
I decided it would be better for readability

Reply:
It's not. Refactor it. You could create some word and cut it into pieces.